### ISSUE TITLE
feature_registration_api

### ISSUE THREAD

**Issue by anastasimars** [anastasimars/ecommerce_v2.0#8]
Add User Registration and Account Activation API:
- develop the user registration API endpoint that requires first name, last name, delivery address, and email address and validate input data
-  implement an email verification system where the registration process completes only after the user clicks on an activation link sent to their email
- send a confirmation email with an activation link
- implement endpoint for activation status update upon clicking the link

**Comment by anastasimars** [comment]
The registration logic will be moved to a separate microservice. A new issue has been created to track this task: [](https://github.com/anastasimars/ecommerce_v2.0/issues/13)
